api/cluster.go

```diff
@@ -363,7 +363,13 @@ func (s *Server) peerQuerySpeculative(ctx context.Context, data cluster.Traceabl
 
 	result := make(map[string]PeerResponse)
 
-	specCheckTicker := time.NewTicker(5 * time.Millisecond)
+	var ticker *time.Ticker
+	var tickChan <-chan time.Time
+	if speculationThreshold != 1 {
```

Comment:
that's nice. so the ticker doesn't even get instantiated if `speculationThreshold == 1`